### PR TITLE
Allow 'is' for static classes for compatibility

### LINKED ISSUE #324: [CodePlex] Error CS7023 'is' for static classes [breaking compiler change]
Originally from [http://roslyn.codeplex.com/workitem/315](http://roslyn.codeplex.com/workitem/315)

@JanKucera reported:
The `ShellProperty<T>` class from `WindowsAPICodePack `stopped compiling in Roslyn.

It uses the `Nullable` class in the way similar to this:

```
class MyClass<T>
{
    public bool IsNullable(T value)
    {
       return value is Nullable;
   }
}
```

which, since `Nullable` is a static class, likely does not do what intended.

Roslyn generates

```
error CS7023: The second operand of an 'is' or 'as' operator may not be static type 'System.Nullable'.
```

However, shouldn't this be treated more as "The result of the expression is always 'false'" warning?

### PR DESCRIPTION
**Pull request by gafter**
but continue to report the diagnostic if "strict" mode is requested.
Fixes #324

### CODE REVIEW

**Comment by gafter** [comment]
Hey, can a couple of you please look at this @jaredpar @AlekseyTs @VladimirReshetnikov @VSadov @agocke

File: src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs [hunk 1]
```diff
@@ -2542,7 +2542,7 @@ private BoundExpression BindIsOperator(BinaryExpressionSyntax node, DiagnosticBa
             // BREAKING CHANGE: called "Referencing Static Class Types".) We break with the native
             // BREAKING CHANGE: compiler here and turn this into an error, as it should be.
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs]: Should this comment be adjusted too?

File: src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs [hunk 2]
```diff
@@ -2932,7 +2932,7 @@ private BoundExpression BindAsOperator(BinaryExpressionSyntax node, DiagnosticBa
             // BREAKING CHANGE: It also allows "someObject as C" if "someObject"
             // BREAKING CHANGE: is of type object.
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs]: Should this comment be adjusted too?

File: src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs [hunk 3]
```diff
@@ -156,6 +156,13 @@ internal override CommonAnonymousTypeManager CommonAnonymousTypeManager
         }
 
         /// <summary>
+        /// True when the compiler is run in "strict" mode, in which it enforces the language specification
+        /// even at the expense of full compatibility. There is a difference when previous versions of the
+        /// compiler failed to enforce the full language specification.
+        /// </summary>
+        internal bool Feature_Strict => Feature("strict") != null;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs]: Does name "Feature_Strict" follow coding style guidelines?

File: src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs (hunk 1 above)
Comment (gafter) [inline src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs]: Yes

File: src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs (hunk 2 above)
Comment (gafter) [inline src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs]: Yes

**Comment by gafter** [comment]
Ping for review, please @jaredpar @AlekseyTs @VladimirReshetnikov @VSadov @agocke

**Comment by AlekseyTs** [comment]
Is the change in Roslyn.sln intentional? Otherwise LGTM.

**Comment by jaredpar** [comment]
LGTM

**Comment by gafter** [comment]
Reverting the change to Roslyn.sln, and squashing the commits.

**Pull request merged**